comictaggerlib/taggerwindow.py

```diff
@@ -1791,36 +1833,40 @@ def identify_and_tag_single_archive(
                 if self.config[0].Issue_Identifier__auto_imprint:
                     md.fix_publisher()
 
-                if not ca.write_metadata(md, self.save_data_style):
-                    match_results.write_failures.append(
-                        Result(
-                            Action.save,
-                            Status.write_failure,
-                            ca.path,
-                            online_results=matches,
-                            match_status=MatchStatus.good_match,
+                # Save each style
+                for style in self.save_data_styles:
+                    if not ca.write_metadata(md, style):
+                        match_results.write_failures.append(
+                            Result(
+                                Action.save,
+                                Status.write_failure,
+                                ca.path,
+                                online_results=matches,
+                                match_status=MatchStatus.good_match,
+                            )
                         )
-                    )
-                    self.auto_tag_log("Save failed ;-(\n")
-                else:
-                    match_results.good_matches.append(
-                        Result(
-                            Action.save,
-                            Status.success,
-                            ca.path,
-                            online_results=matches,
-                            match_status=MatchStatus.good_match,
+                        self.auto_tag_log(f"{metadata_styles[style].name()} save failed ;-(\n")
+                    else:
+                        match_results.good_matches.append(
+                            Result(
+                                Action.save,
+                                Status.success,
+                                ca.path,
+                                online_results=matches,
+                                match_status=MatchStatus.good_match,
+                            )
```

Comment:
This should break the loop. Operating on files that are potentially corrupt can cause further data loss